### PR TITLE
Validate that assigned subtasks in test groups are defined

### LINKED ISSUE #22: Increase the limit of subtasks and test groups to 10

### LINKED ISSUE #114: Prevent test groups from being in nonexistent subtasks

### PR DESCRIPTION
**Pull request by fushar**
Resolves #114

### CODE REVIEW

**Comment by jonathanirvings** [comment]
@fushar You mean this resolves #114, not #22 right?

**Comment by fushar** [comment]
@jonathanirvings Right, fixed :)

File: include/tcframe/spec/core/BaseTestSpec.hpp [hunk 1]
```diff
@@ -182,6 +183,17 @@ class BaseTestSpec : public TProblemSpec, protected TestSuiteBuilder {
     virtual void TestGroup23() {throw NotImplementedException();}
     virtual void TestGroup24() {throw NotImplementedException();}
     virtual void TestGroup25() {throw NotImplementedException();}
+
+private:
+    set<int> getDefinedSubtaskIds(const ConstraintSuite& constraintSuite) const {
```
Comment (jonathanirvings) [inline include/tcframe/spec/core/BaseTestSpec.hpp]: should we make this a public method for ConstraintSuite instead?

File: include/tcframe/util/StringUtils.hpp [hunk 2]
```diff
@@ -33,6 +35,22 @@ class StringUtils {
     }
 
     template<typename T>
+    static string setToString(const set<T>& set) {
+        ostringstream out;
+        out << "{";
+        bool any = false;
+        for (T val : set) {
```
Comment (jonathanirvings) [inline include/tcframe/util/StringUtils.hpp]: `const T&` instead?

File: include/tcframe/spec/core/BaseTestSpec.hpp (hunk 1 above)
Comment (fushar) [inline include/tcframe/spec/core/BaseTestSpec.hpp]: Cool, better idea, done

File: include/tcframe/util/StringUtils.hpp (hunk 2 above)
Comment (fushar) [inline include/tcframe/util/StringUtils.hpp]: Good catch; fixed

**Comment by jonathanirvings** [comment]
LGTM

**Pull request merged**